WebClient/src/communicator.ts

```diff
@@ -187,26 +213,52 @@ export class Communicator implements ICommunicator {
        //this.callbacksByResponder.set(responder, respondCallback);
        // console.log("callbacksByResponder");
        // console.log(this.callbacksByResponder);
-    
-        let correlationID = Guid.create().toString();
-        let requestToSend = new Request(correlationID, additionalData, this.userId, null, responder);
-        console.log(requestToSend);
-        //let serviceTask = this.connection.invoke("QueryAsync", requestToSend).catch(err => console.log(err));
-        let serviceTask = this.connection.invoke("QueryAsync", requestToSend);
-       let timeoutTask = this.timeoutAsync();
 
-        let taskResult = await Promise.race([serviceTask, timeoutTask]);
-        console.log(taskResult);
-        return taskResult;
+
+       let verifyResponder: IResponse = await this.connection.invoke("VerifyResponderIsInList", responder);
+       console.log(verifyResponder.Success);
+       if (verifyResponder.Success) {
+           let correlationID = Guid.create().toString();
+           let requestToSend = new Request(correlationID, additionalData, this.userId, null, responder);
+           console.log(requestToSend);
+           //let serviceTask = this.connection.invoke("QueryAsync", requestToSend).catch(err => console.log(err));
+
+           let serviceTask = this.connection.invoke("QueryAsync", requestToSend);
+           let timeoutTask = this.timeoutAsync();
```

Comment:
Request/respond times out after 2 seconds. But if I extend the time out to 5 seconds, it gets executed correctly and the requester get a response from responder. Still need to figure out where in the process it is taking so long.